irods/test/data_obj_test.py

```diff
@@ -2044,6 +2044,40 @@ def test_append_mode_will_append_to_data_object__issue_495(self):
             if data.exists(testfile):
                 data.unlink(testfile,force=True)
 
+    def test_update_mtime_of_data_object_using_touch_operation_as_non_admin__525(self):
+        try:
+            user_session = self.logins.session_for_user(RODSUSER)
+
+            # Create a data object.
+            data_object_path = f'{helpers.home_collection(user_session)}/test_update_mtime_of_data_object_using_touch_operation__525.txt'
+            self.assertFalse(user_session.data_objects.exists(data_object_path))
+            user_session.data_objects.touch(data_object_path)
+            self.assertTrue(user_session.data_objects.exists(data_object_path))
+
+            # Capture mtime of data object.
+            data_object = user_session.data_objects.get(data_object_path)
+            old_mtime = data_object.replicas[0].modify_time
+
+            # Set the mtime to an earlier time.
+            user_session.data_objects.touch(data_object_path, seconds_since_epoch=1400000000)
+
+            # Compare mtimes for correctness.
+            data_object = user_session.data_objects.get(data_object_path)
+            self.assertGreater(old_mtime, data_object.replicas[0].modify_time)
```

Comment:
Same question here as the collection_manager version of this test...

Reply:
See https://github.com/irods/python-irodsclient/pull/531#discussion_r1598728834.

Reply:
Done.